gap/elements/bipart.gi

```diff
@@ -101,22 +101,26 @@ function(classes)
                   "homogeneous lists,");
   fi;
 
-  if not ForAll(classes, x -> ForAll(x, i -> IsPosInt(i) or IsNegInt(i))) then
+  n := Sum(classes, Length) / 2;
+
+  if n >= 2 ^ 29 then
+    ErrorNoReturn("Semigroups: Bipartition: usage,\n",
+                  "the maximum degree of a bipartition ",
+                  "is 2 ^ 29 - 1,");
+  elif not ForAll(classes, x -> ForAll(x,
+                                       i -> (IsPosInt(i) or IsNegInt(i))
+                                            and AbsInt(i) <= n)) then
     ErrorNoReturn("Semigroups: Bipartition: usage,\n",
                   "the argument <classes> must consist of positive and/or ",
```

Comment:
...must consist of **lists of** positive and/or negative integers... right?

Reply:
Right, thanks will update